### ISSUE TITLE
isDST() always return False

### ISSUE THREAD

**Issue by HenonoaH** [moment/moment#5852]
I have a Node.js app that schedules an email my server Linux box is Ubuntu 18.04.3 LTS and time runs in UTC. I am trying to check Daylight Saving but it fails to false all time! I Googled to sort this issue but my case looks weird am I missing anything here?

`console.log(moment.utc('03:00', 'HH:mm').utcOffset('-05:00').isDST());`

The output of the following consoles,
```
console.log((new Date()).toString());
console.log((new Date()).toLocaleString());
console.log((new Date()).getTimezoneOffset());
console.log(moment.version);
```
![image](https://user-images.githubusercontent.com/32451863/111200251-89286e80-85e7-11eb-9e25-6c0565e75676.png)

**Comment by gilmoreorless** [comment]
Daylight saving is only applicable when a time zone changes which UTC offset it's using. But by definition, hard-coded offsets such as UTC or `-05:00` only have one fixed offset, so there can't be any DST. Therefore always returning `false` for those cases is the correct behaviour.

**Comment by BorBorBor** [comment]
@gilmoreorless There is a bug though. We don't use a fixed offset (but rather `.tz('Europe/Oslo')`) in an existing project. This always worked well until last year, when it stopped switching between DST. Since then I've been manually updating the source code every half year until this issue is fixed…

**Comment by Spacarar** [comment]
@BorBorBor shouldn't you be using `moment.utc('03:00', 'HH:mm').tz('Europe/Oslo').isDST()` instead? It seems obvious that casting it using `.utcOffset('-05:00')` would not ever return `isDST() == true` because it's not a timezone but rather a specific offset you're asking for.

offsets do not use DST, timezones do

**Comment by BorBorBor** [comment]
@Spacarar Yes, as I said we *are* using a timezone and *not* a fixed offset… This is a code snippet:
`              const startofweek =`
`                moment(pstart)`
`                .tz('Europe/Oslo')`
`                .startOf('day')`
`                .subtract(pstart.day(), 'days')`
`              const startofweekdst = startofweek.isDST()`

It worked for roughly 4 years and then it stopped working, the last few years we've manually changed it to and from `const startofweekdst = false`. Which is a terrible way to 'fix' this.

**Comment by Spacarar** [comment]
@BorBorBor
I'd first like to note that `.startOf('day').subtract(pstart.day, 'days')` is just a confusing way of saying `startOf('week')`

I was able to run this in the browser and get true. it appears that DST in oslo is from march 27th to october 30th which is different from my personal DST of march 13th to november 13th.

```
let pstart = moment('2022-05-01')
let startofweek = moment(pstart).tz('Europe/Oslo').startOf('week')
console.log('is dst', startofweek.isDST())
```

I don't think this is an issue with moment.

**Comment by BorBorBor** [comment]
@Spacarar Thanks for taking the time to help me here. To be clear, this is a project I inherited so I don't have context for all the design choices made. Also, I'm indeed in the Oslo time zone where DST started last week. 

I've looked through my old commits and the issue seems to be opposite to what I remembered — that it never became `false` in winter time. I'm able to test it with the console as you say and it does seem to return false if I set a date in March, so I'm not fully sure what has happened on our side for it not to work anymore. Our code is a bit spaghetti-esque though. I will probably try to find some time this summer to find out what is happening here and either remedy the problem, or get rid of Moment alltogether since it's (kind of) EOL now and we probably don't need it anymore in this project.

Thanks again!

**Comment by Spacarar** [comment]
@HenonoaH  Since this was determined not to be an issue with moment. Would you mind closing the issue?

**Comment by anthonyz-fb** [comment]
@Spacarar If you try moment().tz('Europe/Oslo').isDST() and the current date is in DST it will not work...

**Comment by Spacarar** [comment]
@anthonyz-fb your code sample does not show a complete picture of what you've done. Oslo is +1 or +2 depending on the time of year, so this looks like it could not have been Oslo as you suggested.

In testing briefly it appears to be working as intended. `moment('2020-10-13').tz('Europe/Oslo').isDST()` returns true as expected and `moment('2020-11-01').tz('Europe/Oslo').isDST()` returns false as expected. 

![Screen Shot 2022-10-04 at 4 38 33 PM](https://user-images.githubusercontent.com/24302366/193943524-7f4c9aca-0e8c-4bde-85d5-e63c5128b9ef.png)


looking up the DST times for Oslo in 2020 you can see it is expected to be in DST from March 29th to October 30th
![Screen Shot 2022-10-04 at 4 33 26 PM](https://user-images.githubusercontent.com/24302366/193943388-20512757-32ac-46de-b3b2-80917103e539.png)

Do you have a more complete example of what you believe is incorrect?

**Comment by notanthony** [comment]
@Spacarar 
Yeah sorry I was not clear. That was just me testing it using my time zone ‘America/Toronto’ 

Try running these two lines of code:
`moment().tz('America/Toronto').isDST()  

moment(‘2022-9-13).tz('America/Toronto').isDST() `

moment() uses the current time, which is in DST, but it will return false. Like I showed in the image above, when you print it as a string the offset is -5 GMT but when you put it was a date the offset is -4 GMT which I believe is the source of the bug.

**Comment by anthonyz-fb** [comment]
@Spacarar Hey, it worked today. I am not sure what I was doing yesterday but I guess its fine now, whoops

**Comment by jorgeriv** [comment]
I'm experiencing this issue, DST for America/New_York starts on the second Sunday in March and ends first Sunday in November however moment returns false for any date in that range (or outside the range which is ok).
<img width="449" alt="Screenshot 2023-07-27 at 9 11 06" src="https://github.com/moment/moment/assets/528606/5e55d62f-fc60-46ca-a00c-9bee6a76b4bf">

**Comment by Spacarar** [comment]
@jorgeriv I'm not seeing the same results running this in the console
![Screenshot 2023-07-27 at 9 24 17 AM](https://github.com/moment/moment/assets/24302366/213b90ba-daa1-4cfe-a4ac-1ae268e350ae)

**Comment by jorgeriv** [comment]
@Spacarar It works fine when I try to create a minimum reproducible example in stackblitz 👍 . eg.

```javascript
const moment = require('moment-timezone');

const isDST = moment('2023-06-10T12:30').tz('America/New_York').isDST();

console.log(isDST);
```

But the exact same does not work on our project for me or any of my coworkers. I don't know what kind of sorcery is that. Maybe something else some how is interfering with moment inside the larger project, but I don't know what could be it. If I find something I'll post here.

**Comment by gilmoreorless** [comment]
@jorgeriv Check if your project is importing a file called something like `moment-timezone-with-data-2012-2022.js`. As the filename suggests, it doesn't contain data for 2023 and beyond. The file is now deprecated and will be removed at some point, mainly because it causes issues just like this.
The equivalent replacement is a file called `moment-timezone-with-data-10-year-range.js`, which gets an updated date range with each release. (See the ["Where to use it" docs](https://momentjs.com/timezone/docs/#/use-it/) for details of the files available.)

**Comment by jorgeriv** [comment]
@gilmoreorless Sorry for the late response, just made the change you suggested and this fixed the issue. I wanted to confirm in case someone else encounter this same error and reads this thread. Thank you very much!

**Comment by BorBorBor** [comment]
@gilmoreorless no fucking way, that is also the issue in my instance. Have been setting and resetting DST manually for almost three years now :')

**Comment by mattjohnsonpint** [comment]
Thanks for the report and all the follow-up testing! A fixed offset such as UTC or `-05:00` has no DST rules, so `isDST()` correctly returns `false`; a named zone is required. I also appreciate the confirmations that later failures came from the expired `moment-timezone-with-data-2012-2022.js` bundle and were fixed by using current data. I'm closing this since both causes are confirmed.
